match.py

```diff
@@ -0,0 +1,173 @@
+        if ip_int & mask != flow_ip_int & mask:
+            return False
+    if not wildcards & FlowWildCards.OFPFW_NW_TOS:
+        if 'ip_tos' not in args:
+            return False
+        if match_fields.get('ip_tos') != int(args.get('ip_tos')):
+            return False
+    if not wildcards & FlowWildCards.OFPFW_NW_PROTO:
+        if 'ip_proto' not in args:
+            return False
+        if match_fields.get('ip_proto') != int(args.get('ip_proto')):
+            return False
+    if not wildcards & FlowWildCards.OFPFW_TP_SRC:
+        if 'tp_src' not in args:
+            return False
+        if match_fields.get('tcp_src') != int(args.get('tp_src')):
+            return False
+    if not wildcards & FlowWildCards.OFPFW_TP_DST:
+        if 'tp_dst' not in args:
+            return False
+        if match_fields.get('tcp_dst') != int(args.get('tp_dst')):
+            return False
+    return True
+
+
+# pylint: disable=too-many-return-statements, too-many-statements, R0912
+def match10(flow_dict, args):
+    """Match a packet against this flow (OF1.0)."""
+    match_fields = _get_match_fields(flow_dict)
+    wildcards = match_fields.get('wildcards')
+    if not wildcards & FlowWildCards.OFPFW_IN_PORT:
+        if 'in_port' not in args:
+            return False
+        if match_fields.get('in_port') != int(args.get('in_port')):
+            return False
+    if not wildcards & FlowWildCards.OFPFW_DL_VLAN_PCP:
+        if 'vlan_pcp' not in args:
+            return False
+        if match_fields.get('vlan_pcp') != int(args.get('vlan_pcp')):
+            return False
+    if not wildcards & FlowWildCards.OFPFW_DL_VLAN:
+        if 'vlan_vid' not in args:
+            return False
+        if match_fields.get('vlan_vid') != args.get('vlan_vid')[-1]:
+            return False
+    if not wildcards & FlowWildCards.OFPFW_DL_SRC:
+        if 'eth_src' not in args:
+            return False
+        if match_fields.get('eth_src') != args.get('eth_src'):
+            return False
+    if not wildcards & FlowWildCards.OFPFW_DL_DST:
+        if 'eth_dst' not in args:
+            return False
+        if match_fields.get('eth_dst') != args.get('eth_dst'):
+            return False
+    if not wildcards & FlowWildCards.OFPFW_DL_TYPE:
+        if 'eth_type' not in args:
+            return False
+        if match_fields.get('eth_type') != int(args.get('eth_type')):
+            return False
+    if not _match_ipv4_10(match_fields, args, wildcards):
+        return False
+    return flow_dict
+
+
+def match13_no_strict(flow_to_install, stored_flow_dict):
+    """Match a packet againts the stored flow (OF 1.3).
+
+    Return the flow if any fields match, otherwise, return False.
+    """
+    if flow_to_install.get('cookie_mask') and 'cookie' in stored_flow_dict:
+        cookie = flow_to_install['cookie_mask'] & flow_to_install['cookie']
+        if cookie == stored_flow_dict['cookie']:
+            return stored_flow_dict
+        return False
```

Comment:
This does not seem to be right.
Let's take, for example, an IPv4 (it is easier to visualize).
In `flow_to_install` we have:

```
ip_addr = 192.168.1.1
ip_mask = 255.255.255.0
```

And in `stored_flow_dict` we have:
```
ip_addr = 192.168.1.120
```
Clearly, the address in `stored_flow_dict` matches the address/mask in `flow_to_install`, but the code would
compare `if "192.168.1.0" == "192.168.1.120"`.
To check if the given IP is in the network, it is necessary to apply the mask to both IPs, and then compare the results.
The same applies to the cookie field.

```suggestion
        cookie = flow_to_install['cookie_mask'] & flow_to_install['cookie']
        stored_cookie = stored_flow_dict['cookie'] & flow_to_install['cookie']
        if cookie == stored_cookie:
            return stored_flow_dict
        return False
```

Reply:
Resolved in https://github.com/kytos/flow_manager/pull/117/commits/7a5e38bf8ee96e0ee7cc049931ed30e736f1c3f0